src/app/core/store/documents/documents.effects.ts

```diff
@@ -250,6 +253,31 @@ export class DocumentsEffects {
     })
   );
 
+  @Effect()
+  public checkDataHint$: Observable<Action> = this.actions$.pipe(
+    ofType<DocumentsAction.CheckDataHint>(DocumentsActionType.CHECK_DATA_HINT),
+    withLatestFrom(
+      this.store$.pipe(select(selectDocumentsDictionary)),
+      this.store$.pipe(select(selectCollectionsDictionary))
+    ),
+    mergeMap(([action, documents, collections]) => {
+      const document = action.payload.document;
+      const myDocuments = Object.values(documents).filter(d => d.collectionId === document.collectionId);
+      const myCollection = collections[document.collectionId];
```

Comment:
I am not sure about naming variables like that. Please change it to something meaningful.